tests/unit/dojson/test_dojson_hep_bd01x09x.py

```diff
@@ -507,49 +497,56 @@ def test_texkeys_from_multiple_035__a_z_9():
             <subfield code="a">Akiba:2016ofq</subfield>
           </datafield>
         </record>
-    ''' # record/1498308
+    '''  # record/1498308
 
     # the first is the one containing the a, the rest is from the z
-    expected_head = ['Akiba:2016ofq', 'N.Cartiglia:2015cn']
+    expected = [
+        'Akiba:2016ofq',
+        'N.Cartiglia:2015cn',
+    ]
     result = hep.do(create_record(snippet))
 
     assert validate(result['texkeys'], subschema) is None
     assert expected == result['texkeys']
-    assert [{}] == result['external_system_identifiers']
 
     expected = [
-        {'9': 'INSPIRETeX', 'a': 'Akiba:2016ofq'},
-        {'9': 'INSPIRETeX', 'z': 'N.Cartiglia:2015cn'}
+        {
+            '9': 'INSPIRETeX',
+            'a': 'Akiba:2016ofq',
+        },
+        {
+            '9': 'INSPIRETeX',
+            'z': 'N.Cartiglia:2015cn',
+        }
     ]
     result = hep2marc.do(result)
 
-    assert sorted(expected) == sorted(result['035'])
+    assert expected == result['035']
 
 
-@pytest.mark.xfail(reason='Schema 13')
 def test_discard_035__9_arXiv():
     '''035__9:arXiv is redundant with 037__9:arXiv, throw it away'''
     schema = load_schema('hep')
+    subschema = schema['properties']['external_system_identifiers']
 
     snippet = '''
         <datafield tag="035" ind1=" " ind2=" ">
           <subfield code="9">arXiv</subfield>
           <subfield code="a">oai:arXiv.org:1611.05079</subfield>
         </datafield>
-    ''' # record/1498308
+    '''  # record/1498308
 
-    expected = [{}]
     result = hep.do(create_record(snippet))
 
-    assert validate(result, schema) is None
-    assert expected == result['external_system_identifiers']
-    assert expected == result['arxiv_eprints']
+    try:
+        validate(result.get('external_system_identifiers'), subschema)
+    except ValidationError:
+        assert True
```

Comment:
This is not how you assert that something raises. Here's how you do it: http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions